tls/extensions/s2n_client_supported_versions.c

```diff
@@ -83,12 +81,15 @@ int s2n_extensions_client_supported_versions_process(struct s2n_connection *conn
             continue;
         }
 
+        conn->client_protocol_version = MAX(client_version, conn->client_protocol_version);
```

Comment:
this may prevent us from recording what the true maximum client protocol version is, at least in the case of with #1623, supported versions can still be parsed when client_version is higher than the version we are negotiating

Reply:
It certainly does. I see you are checking that the client protocol doesn't change in some tests in #1623 though. The RFC says we shouldn't do any unknown versions anyway. I'll update this.